universal-application-tool-0.0.1/app/services/program/ProgramServiceImpl.java

```diff
@@ -29,13 +41,25 @@ public ProgramServiceImpl(ProgramRepository programRepository) {
 
   @Override
   public CompletionStage<ImmutableList<ProgramDefinition>> listProgramDefinitionsAsync() {
-    return programRepository
-        .listPrograms()
-        .thenApply(
-            programs ->
-                programs.stream()
-                    .map(program -> program.getProgramDefinition())
-                    .collect(ImmutableList.toImmutableList()));
+    CompletableFuture<ReadOnlyQuestionService> roQuestionServiceFuture =
+        questionService.getReadOnlyQuestionService().toCompletableFuture();
+    CompletableFuture<List<Program>> programsFuture =
+        programRepository.listPrograms().toCompletableFuture();
+
+    return CompletableFuture.allOf(roQuestionServiceFuture, programsFuture)
+        .thenApplyAsync(
+            ignore -> {
+              ReadOnlyQuestionService roQuestionService = roQuestionServiceFuture.join();
+              List<Program> programs = programsFuture.join();
```

Comment:
Suggestion (non-blocking): I know `ProgramRepository` is written to return a `List` so it may be out of scope, but our dev standards stipulate using immutable collections whenever possible. Please either file an issue and add a TODO to change the repo return type to immutable or do it in this PR.

Reply:
I was gonna fix this in a follow-up PR. I'll add a TODO.